presentation/json.go

```diff
@@ -0,0 +1,140 @@
+package presentation
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/box/memsniff/analysis"
+	"os"
+	"time"
+)
+
+func (u *uiContext) runLoggingEventLoop() error {
+	updateTick := time.NewTicker(u.interval)
+	defer updateTick.Stop()
+	if err := u.update(); err != nil {
+		return err
+	}
+
+	for {
+		select {
+		case <-updateTick.C:
+			if err := u.updateReport(); err != nil {
+				return err
+			}
+
+		case msg := <-u.msgChan:
+			u.logger.Log(msg)
+		}
+	}
+}
+
+func (u *uiContext) updateReport() error {
+	rep := u.analysis.Report(!u.cumulative)
+	rep.SortBy(-2)
+
+	numKeysSeen := len(rep.Rows)
+	totalBandwidthUsed := totalBytesUseForKeys(rep.Rows)
+
+	s := u.statProvider()
+
+	u.truncateResultsToMaxAndTopX(&rep)
+	u.prevReport = rep
+	numReportedKeys := len(rep.Rows)
+	reportedKeysBandwidth := totalBytesUseForKeys(rep.Rows)
+
+	f, err := u.openFile(u.outputFile)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	reportJson, _ := formatReportAsJson(u.prevReport, s, numKeysSeen, totalBandwidthUsed,numReportedKeys,reportedKeysBandwidth)
```

Comment:
There should be spaces between arguments. You need to run this through `go fmt`.